src/main/java/it/gov/pagopa/pu/registry/dto/RegistryEventDTO.java

```diff
@@ -0,0 +1,78 @@
+package it.gov.pagopa.pu.registry.dto;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import it.gov.pagopa.pu.registry.enums.RegistryEventSubType;
+import it.gov.pagopa.pu.registry.enums.RegistryEventType;
+import jakarta.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+import java.time.OffsetDateTime;
+
+@Data
+@SuperBuilder
+@NoArgsConstructor
+@AllArgsConstructor
+@JsonTypeInfo(
+  use = JsonTypeInfo.Id.NAME,
+  include = JsonTypeInfo.As.EXISTING_PROPERTY,
+  property = "eventType",
+  defaultImpl = RegistryEventDTO.class,
+  visible = true
+)
+@JsonSubTypes({
+  @JsonSubTypes.Type(value = RegistryEventPagoPaDTO.class, names = {
+    "paVerifyPaymentNotice",
+    "paGetPaymentV2",
+    "paSendRTV2",
+    "newDebtPosition",
+    "createPosition",
+    "updatePosition",
+    "deletePosition",
+    "fetchPaymentReporting"
+  }),
+  @JsonSubTypes.Type(value = RegistryEventSilDTO.class, names = {
+    "paaSILAutorizzaImportFlusso",
+    "paaSILChiediStatoImportFlusso",
+    "paaSILImportaDovuto",
+    "paaSILPrenotaExportFlusso",
+    "paaSILPrenotaExportFlussoIncrementaleConRicevuta",
+    "paaSILChiediStatoExportFlusso",
+    "paaSILInviaDovuti",
+    "paaSILVerificaAvviso",
+    "paaSILChiediPagati",
+    "paaSILChiediPagatiConRicevuta",
+    "paaSILInviaCarrelloDovuti",
+    "paaSILChiediEsitoCarrelloDovuti",
+    "pivotSILAutorizzaImportFlusso",
+    "pivotSILChiediStatoImportFlusso",
+    "pivotSILAutorizzaImportFlussoTesoreria",
+    "pivotSILChiediStatoImportFlussoTesoreria",
+    "pivotSILPrenotaExportFlussoRiconciliazione",
+    "pivotSILChiediStatoExportFlussoRiconciliazione",
+    "pivotSILChiediAccertamento",
+    "attualizzazioneImporti",
+    "notificaPagamento"
+  })
+})
+public class RegistryEventDTO {
```

Comment:
not a good name!
this application is handling also DebtPositionRegistry and InstallmentRegistry: one could think this should be used also there...
```suggestion
public class RegistryInterfaceEventDTO {
```